File: donors/orchestration-java/ValidationContract.java
package com.aiassistant.core.orchestration;

import java.util.Map;

public interface ValidationContract {
    
    ValidationResult validate(String componentId, Map<String, Object> output);
    
    public static class ValidationResult {
        public final boolean isValid;
        public final String message;
        public final Map<String, Object> details;
        
        public ValidationResult(boolean isValid, String message, Map<String, Object> details) {
            this.isValid = isValid;
            this.message = message;
            this.details = details;
        }
        
        public static ValidationResult success() {
            return new ValidationResult(true, "Validation passed", null);
        }
        
        public static ValidationResult failure(String message) {
            return new ValidationResult(false, message, null);
        }
        
        public static ValidationResult failure(String message, Map<String, Object> details) {
            return new ValidationResult(false, message, details);
        }
    }
}

File: android/app/src/main/kotlin/com/ariaagent/mobile/core/orchestration/ValidationContract.kt
// Ported from donors/orchestration-java/ValidationContract.java
// Original repo: TITANICBHAI/AI-ASSISTANT-INCOMPLETE
// Changes: idiomatic Kotlin interface + sealed-result-style data class with factories.

package com.ariaagent.mobile.core.orchestration

/**
 * Optional contract a component can provide to the orchestrator that lets the
 * orchestrator validate its outputs before they are forwarded downstream.
 */
fun interface ValidationContract {
    fun validate(componentId: String, output: Map<String, Any?>): Result

    data class Result(
        val isValid: Boolean,
        val message: String,
        val details: Map<String, Any?> = emptyMap(),
    ) {
        companion object {
            fun success(): Result = Result(true, "Validation passed")
            fun failure(message: String, details: Map<String, Any?> = emptyMap()): Result =
                Result(false, message, details)
        }
    }
}
